src/librustc/mir/mod.rs

```diff
@@ -1978,6 +1978,14 @@ impl<'tcx> Place<'tcx> {
             _ => None,
         }
     }
+
+    pub fn base_local(&self) -> Option<Local> {
```

Comment:
Doc comment: 

---


Returns the local variable at the "base" of this place, if any.  e.g., for a place like `a.b.c`, returns `Some(a)`. For a place based in a static or constant, returns `None`.